server/src/main/java/de/uftos/repositories/solver/timefold/constraints/ConstraintDefinitionFactory.java

```diff
@@ -0,0 +1,1073 @@
+      Set<ResourceTimefoldInstance> filteredSet = new HashSet<>();
+      outerLoop:
+      for (ResourceTimefoldInstance resource : set) {
+        parameters.set(0, resource); //replacing "this"-pointer
+        for (Function<List<ResourceTimefoldInstance>, Boolean> function : functions) {
+          if (!function.apply(parameters)) {
+            continue outerLoop;
+          }
+        }
+        filteredSet.add(resource);
+      }
+
+      parameters.set(0, thisPointer);
+
+      return filteredSet;
+    });
+  }
+
+  private static Function<List<ResourceTimefoldInstance>, Boolean> convertIsEmpty(
+      AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    if (ast.getToken() != UcdlToken.IS_EMPTY) {
+      throw new IllegalStateException();
+    }
+    OperatorDto isEmpty = (OperatorDto) ast;
+    if (isEmpty.parameters().size() != 1) {
+      throw new IllegalStateException();
+    }
+    Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>> set =
+        convertSet(isEmpty.parameters().getFirst(), params);
+    return (parameters) -> set.apply(parameters).isEmpty();
+  }
+
+  private static Function<List<ResourceTimefoldInstance>, Number> convertSize(
+      AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    if (ast.getToken() != UcdlToken.SIZE) {
+      throw new IllegalStateException();
+    }
+    OperatorDto size = (OperatorDto) ast;
+    if (size.parameters().size() != 1) {
+      throw new IllegalStateException();
+    }
+    Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>> set =
+        convertSet(size.parameters().getFirst(), params);
+    return (parameters) -> new Number(set.apply(parameters).size());
+  }
+
+  private static Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>> convertSet(
+      AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    return switch (ast.getToken()) {
+      case RESOURCE_SET -> convertResourceSet(ast, params);
+      case NUMBER_SET -> convertNumberSet(ast, params);
+      default -> throw new IllegalStateException();
+    };
+  }
+
+  private static Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>>
+      convertResourceSet(AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    if (ast.getToken() != UcdlToken.RESOURCE_SET) {
+      throw new IllegalStateException();
+    }
+    SetDto set = (SetDto) ast;
+
+    Function<List<ResourceTimefoldInstance>, ResourceTimefoldInstance> name;
+
+    if (set.setName().getToken() == UcdlToken.VALUE_REFERENCE) {
+      name = convertValueReference(set.setName(), params);
+    } else {
+      name = convertElement(set.setName(), params);
+    }
+
+    List<Function<
+        List<ResourceTimefoldInstance>,
+        Function<Set<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>>
+        >> modifiers = new ArrayList<>();
```

Comment:
This formatting is very much screwed up. Can you somehow make it better readable?

Reply:
I can try, but i can't promise anything.